lib/videojs_rails/tags/source.rb

```diff
@@ -2,37 +2,28 @@ module VideojsRails
   module Tags
     class Source < Tag
       def initialize(video_type, options)
-        @video_type = video_type
-        parse_options(options)
+        @attributes = prepare_attributes(video_type, options)
       end
 
       def to_html
-        tag :source, attributes
+        tag :source, @attributes
```

Comment:
Why preparing in initializer `@attributes`? You can do it on demand when you generate HTML.